lineapy/utils/config.py

```diff
@@ -125,40 +145,60 @@ def set(self, key: str, value: Any, verbose=True) -> None:
                 self.__dict__[key] = value
                 os.environ[f"LINEAPY_{key.upper()}"] = str(value)
 
+            # Send a heartbeat to artifact_storage_dir
+            if key == "artifact_storage_dir":
+                with fsspec.open(
+                    str(self.safe_get("artifact_storage_dir")).rstrip("/")
+                    + "/heartbeat",
+                    "w",
+                ) as f:
+                    f.write(
+                        datetime.datetime.utcnow().strftime("%Y%m%d%H%M%S")
+                    )
+
     def _set_defaults(self):
         """Fill empty configuration items"""
         self.safe_get("logging_file")
         self.safe_get("database_url")
         self.safe_get("artifact_storage_dir")
         self.safe_get("customized_annotation_folder")
 
-    def safe_get(self, name) -> Union[Path, str]:
-        def safe_get_folder(name) -> Path:
-            """Return folder as pathlib.Path
+    def safe_get(self, name) -> FilePath:
+        def safe_get_folder(name) -> FilePath:
+            """Return folder as FilePath
             Create the folder if it doesn't exist"""
-            if not Path(self.__dict__[name]).exists():
-                logger.warning(
-                    f"Folder {Path(self.__dict__[name]).as_posix()} does not exist. Creating a new one."
-                )
-                Path(self.__dict__[name]).mkdir(parents=True, exist_ok=True)
-            return Path(self.__dict__[name])
+
+            if isinstance(self.__dict__[name], Path) or isinstance(
+                fsspec.core.url_to_fs(self.__dict__[name])[0],
+                fsspec.implementations.local.LocalFileSystem,
```

Comment:
this changed from the last time. are we restricting ourselves to just local here? does this work with s3 as well?